sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -506,13 +533,24 @@ public void teardown() throws Exception {
     void setParameters(T element, PreparedStatement preparedStatement) throws Exception;
   }
 
+  /**
+   * An interface used to detect deadlock when a {@link SQLException} occurs.
+   * If {@link DeadlockPredicate#test(SQLException)} returns true, {@link Write} tries
+   * to replay the statements.
+   */
+  @FunctionalInterface
+  public interface DeadlockPredicate extends Serializable {
```

Comment:
Maybe call this `RetryStrategy` in case the user wants to retry more kinds of `SQLException`s (though by default we should retry only deadlocks), and rename `test` to `apply`? (accordingly updating documentation)

Reply:
+1 great idea !